network.go

```diff
@@ -61,7 +61,7 @@ type NetworkController interface {
 	Networks() []Network
 
 	// WalkNetworks uses the provided function to walk the Network(s) managed by this controller.
-	WalkNetworks(walker NetworkWalker)
+	WalkNetworks(walker NetworkWalker) bool
```

Comment:
I don't think this is needed.

The job of WalkNetworks() is to simply execute the client's NetworkWalker function in multi-thread safe fashion without exposing the internal data structures. And do this in the most generic way, so that no constraints are put on the intent or implementation of the walking function.

The implementation of the walker function is totally up to the client. If WalkNetworks had to return the bool received from the NetworkWalker, then the client function would already know about it.